### PR TITLE
918 - OASys sections selections endpoint

### PR DESCRIPTION
**Pull request by brownbearsoftware**

### CODE REVIEW

**Comment by brownbearsoftware** [comment]
[OASys sections selection endpoint](https://trello.com/c/yXbAULtJ/918-oasys-sections-selection-endpoint)

**Comment by edavey** [comment]
Thanks, but I realise on reviewing this that the design I proposed of grouping into:

- linked to harm
- linked to reoffending
- linked to neither

doesn't easily cover the case where a sections is linked to *both*.

Sorry -- so as discussed let's go back to a simpler (front-end agnostic) design where these 2 booleans are properties on each section, e.g.

```json
{
  "section": 3,
  "name": "Accommodation",
  "linkedToHarm": false,
  "linkedToReoffending": true
},
{
  "section": 6,
  "name": "Relationships",
  "linkedToHarm": false,
  "linkedToReoffending": false
},
``` 

Does that look correct and more useful?

**Comment by brownbearsoftware** [comment]
> Thanks, but I realise on reviewing this that the design I proposed of grouping into:
> 
>     * linked to harm
> 
>     * linked to reoffending
> 
>     * linked to neither
> 
> 
> doesn't easily cover the case where a sections is linked to _both_.
> 
> Sorry -- so as discussed let's go back to a simpler (front-end agnostic) design where these 2 booleans are properties on each section, e.g.
> 
> ```json
> {
>   "section": 3,
>   "name": "Accommodation",
>   "linkedToHarm": false,
>   "linkedToReoffending": true
> },
> {
>   "section": 6,
>   "name": "Relationships",
>   "linkedToHarm": false,
>   "linkedToReoffending": false
> },
> ```
> 
> Does that look correct and more useful?

Makes sense, have updated accordingly

File: src/main/resources/static/api.yml [hunk 1]
```diff
@@ -1191,6 +1191,39 @@ paths:
                 $ref: '#/components/schemas/Problem'
         500:
           $ref: '#/components/responses/500Response'
+  /people/{crn}/oasys/selection:
+    get:
+      tags:
+        - OASys
+      summary: Returns the importable sections of OASys, grouped by whether they are linked to harm/reoffending
```
Comment (edavey) [inline src/main/resources/static/api.yml:1198]: I think we need a new summary now? Something like "Returns the importable sections of OASys including details of links to harm and reoffending"?

**Review by edavey — APPROVED** [review]
This looks ideal 👍 
Just a tiny point on the OpenAPI summary which I think needs updating...

**Pull request merged**